# EDA

### Step 1: Problem statement and data collection

In [40]:
import pandas as pd

total_data = pd.read_csv('/workspaces/JLL_Linear_Regression_Regilarized/data/raw/health_insurance_data.csv', sep=',')
total_data.head()

fips  TOT_POP    0-9  0-9 y/o % of total pop  19-Oct  \
0  1001    55601   6787               12.206615    7637   
1  1003   218022  24757               11.355276   26913   
2  1005    24881   2732               10.980266    2960   
3  1007    22400   2456               10.964286    2596   
4  1009    57840   7095               12.266598    7570   

   10-19 y/o % of total pop  20-29  20-29 y/o % of total pop  30-39  \
0                 13.735364   6878                 12.370281   7089   
1                 12.344167  23579                 10.814964  25213   
2                 11.896628   3268                 13.134520   3201   
3                 11.589286   3029                 13.522321   3113   
4                 13.087828   6742                 11.656293   6884   

   30-39 y/o % of total pop  ...  COPD_number  diabetes_prevalence  \
0                 12.749771  ...         3644                 12.9   
1                 11.564429  ...        14692                 12.0   
2                 12.865239  ...         2373                 19.7   
3                 13.897321  ...         1789                 14.1   
4                 11.901798  ...         4661                 13.5   

   diabetes_Lower 95% CI  diabetes_Upper 95% CI  diabetes_number  \
0                   11.9                   13.8             5462   
1                   11.0                   13.1            20520   
2                   18.6                   20.6             3870   
3                   13.2                   14.9             2511   
4                   12.6                   14.5             6017   

   CKD_prevalence  CKD_Lower 95% CI  CKD_Upper 95% CI  CKD_number  \
0             3.1               2.9               3.3        1326   
1             3.2               3.0               3.5        5479   
2             4.5               4.2               4.8         887   
3             3.3               3.1               3.6         595   
4             3.4               3.2               3.7        1507   

   Urban_rural_code  
0                 3  
1                 4  
2                 6  
3                 2  
4                 2  

[5 rows x 108 columns]

### Step 2: Exploration and data cleaning

In [41]:
total_data.info(verbose=True, show_counts=True)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3140 entries, 0 to 3139
Data columns (total 108 columns):
 #    Column                                                                         Non-Null Count  Dtype  
---   ------                                                                         --------------  -----  
 0    fips                                                                           3140 non-null   int64  
 1    TOT_POP                                                                        3140 non-null   int64  
 2    0-9                                                                            3140 non-null   int64  
 3    0-9 y/o % of total pop                                                         3140 non-null   float64
 4    19-Oct                                                                         3140 non-null   int64  
 5    10-19 y/o % of total pop                                                       3140 non-null   float64
 6    20-29         

*Once we have obtained this information, it is important that we are able to draw conclusions, such as the following:

-There are a total of 3140 rows (in this case, people) and 108 columns, among which we will find the target or class to predict later   
-We don´t have nulls values     
-The data has 106 numerical characteristics and 2 categorical characteristics.

***Eliminates Duplicates***

In [27]:
total_data.drop('fips', axis = 1).duplicated().sum()

0

*There aren´t duplicates*

*If there were duplicates:*

In [ ]:
# total_data = total_data.drop_duplicates(subset = total_data.columns.difference([''])) --> Column with ID
# print(total_data.shape)
# total_data.head()

***Eliminate irrelevant information***

In [39]:
total_data[total_data.columns[45:60]]

Bachelor's degree or higher 2014-18  \
0                                   10291   
1                                   46075   
2                                    2220   
3                                    1813   
4                                    5010   
...                                   ...   
3135                                 6243   
3136                                 9847   
3137                                 1995   
3138                                 1220   
3139                                  961   

      Percent of adults with less than a high school diploma 2014-18  \
0                                                  11.3                
1                                                   9.7                
2                                                  27.0                
3                                                  16.8                
4                                                  19.8                
...                                                 ...                
3135                                                9.0                
3136                                                5.6                
3137                                                7.2                
3138                                               10.4                
3139                                                7.8                

      Percent of adults with a high school diploma only 2014-18  \
0                                                  32.6           
1                                                  27.6           
2                                                  35.7           
3                                                  47.3           
4                                                  34.0           
...                                                 ...           
3135                                               33.3           
3136                                               13.2           
3137                                               40.1           
3138                                               30.3           
3139                                               35.8           

      Percent of adults completing some college or associate's degree 2014-18  \
0                                                  28.4                         
1                                                  31.3                         
2                                                  25.1                         
3                                                  24.4                         
4                                                  33.5                         
...                                                 ...                         
3135                                               35.7                         
3136                                               23.8                         
3137                                               37.2                         
3138                                               37.7                         
3139                                               37.3                         

      Percent of adults with a bachelor's degree or higher 2014-18  \
0                                                  27.7              
1                                                  31.3              
2                                                  12.2              
3                                                  11.5              
4                                                  12.6              
...                                                 ...              
3135                                               22.0              
3136                                               57.4              
3137                                               15.4              
3138                                               21.5              
3139                                               1